components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -1507,25 +1507,51 @@ def getRetvalDeclarationForType(returnType, descriptorProvider):
                     returnType)
 
 
-def MemberCondition(pref, func):
+def MemberCondition(pref, func, exposed):
     """
     A string representing the condition for a member to actually be exposed.
     Any of the arguments can be None. If not None, they should have the
     following types:
 
     pref: The name of the preference.
     func: The name of the function.
+    exposed: One or more names of an exposed member.
     """
     assert pref is None or isinstance(pref, str)
     assert func is None or isinstance(func, str)
-    assert func is None or pref is None
+    assert exposed is None or isinstance(exposed, str) or isinstance(exposed, list)
+    assert func is None or pref is None or exposed is None
     if pref:
         return 'Condition::Pref("%s")' % pref
     if func:
         return 'Condition::Func(%s)' % func
+    if exposed:
+        if isinstance(exposed, str):
+            return "Condition::Exposed(%s)" % ExposedCondition(exposed)
     return "Condition::Satisfied"
 
 
+def ExposedCondition(cond):
```

Comment:
This function is just temporary. I wans't sure if there was a better way to get the mappings for the global bit sets. Also, what do 'Worklet' and 'Worker' map to? There is no obvious global bit set for those versus something like 'ServiceWorker'

Reply:
Worklet and Worker map to any interface that has a `[Global=(Worker)]` or `[Global=(Worklet)]` annotation, respectively. For example, both ServiceWorkerGlobalScope and DedicatedWorkerGlobalScope should match Worker.